src/append_only_data.rs

```diff
@@ -663,6 +663,28 @@ where
     P: Permissions + std::hash::Hash + Clone,
 {
     fn append(&mut self, entries: &[(Vec<u8>, Vec<u8>)], last_entries_index: u64) -> Result<()> {
+        let existing: BTreeSet<Vec<u8>> = self
+            .inner
+            .data
+            .iter()
+            .map(|tuple| tuple.0.clone())
+            .collect();
+        let new: BTreeSet<Vec<u8>> = entries.iter().map(|tuple| tuple.0.clone()).collect();
+
+        // If duplicate entries are present in the push.
+        if new.len() < entries.len() {
+            return Err(Error::DuplicateEntries);
+        }
+
+        if !existing.is_disjoint(&new) {
+            let dup: Entries = entries
```

Comment:
If we do take `entries` in as a `mut Vec`, we can avoid cloning here and just do:
```rust
let dup = entries
    .drain(..)
    .filter(|(key, _value)| existing.contains(&key))
    .collect();
```
and likewise at the end of the method, we could just do
```rust
self.inner.data.extend(entries);
```